File: app/src/main/java/ru/gb/course1/myapplication/domain/ColorsRepo.java
package ru.gb.course1.myapplication.domain;

import java.util.List;

// Create Read Update Delete
public interface ColorsRepo {
    // Create
    void addColor(ColorEntity colorEntity);

    // Read
    List<ColorEntity> getColors();

    // Update
    // todo

    // Delete
    void deleteItem(String id);
}

File: app/src/main/java/ru/gb/course1/myapplication/domain/ColorsRepo.kt
package ru.gb.course1.myapplication.domain

import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.Single

// Create Read Update Delete
interface ColorsRepo {
    // Create
    fun addColor(colorEntity: ColorEntity)

    // Read
    fun getColors(): List<ColorEntity>
    val colorsObservable: Observable<List<ColorEntity>>
    val colorsSingle: Single<List<ColorEntity>>

    // Update
    // todo

    // Delete
    fun deleteItem(id: String)
}